### ISSUE TITLE
Join missing from query

### ISSUE THREAD

**Issue by jonneroelofs** [kirschbaum-development/eloquent-power-joins#77]
If you have two different relations to the same table and then have a shared relation on those relations, then if you trying querying them you end up missing 1 join in the resulting query. An example will make this much clearer I think:

```php 
 $query =  \App\Models\Event::query()
            ->leftJoinRelationShip('location.planet', [
                'location' => function($join){
                    $join->as('location_alias');
                },
                'planet' => function($join){
                    $join->as('location_planet_alias');
                }
            ])
            ->leftJoinRelationShip('secondaryLocation.planet', [
                'secondaryLocation' => function($join){
                    $join->as('secondary_location_alias');
                },
                'planet' => function($join){
                    $join->as('secondary_location_planet_alias');
                }
           ])
           ;
```
Results in:
```sql
select `events`.* from `events` 
    left join `locations` as `location_alias` on `events`.`location_id` = `location_alias`.`id` 
    left join `planets` as `location_planet_alias` on `location_alias`.`planet_id` = `location_planet_alias`.`id` 
    left join `locations` as `secondary_location_alias` on `events`.`secondary_location_id` = `secondary_location_alias`.`id`
```

The there should also have been another left join to `planets` for the secondary location.

I think it caused by the usage of the $relationJoinCache in the method scopeJoinNestedRelationship.

```php
$relationJoinCache = $index === 0
                ? "{$relation->getQuery()->getModel()->getTable()}.{$index}.{$relationName}"
                : "{$relation->getQuery()->getModel()->getTable()}.{$latestRelation->getModel()->getTable()}.{$relationName}";
```

It uses the {$latestRelation->getModel()->getTable()} (tablename) but I think it should probably be useing the latestRelation's relationname right?

This seems to fix the problem:
```php
$relationJoinCache = $index === 0
                ? "{$relation->getQuery()->getModel()->getTable()}.{$index}.{$relationName}"
                : "{$relation->getQuery()->getModel()->getTable()}.{$latestRelation->getRelationName()}.{$relationName}";
```

Is this something you can change / would merge if I create a Pr?

**Comment by luisdalmolin** [comment]
@jonneroelofs Thanks for the detailed issue. I was on vacation so took me a while to see this, but I'm definitely open to a PR with a fix to this. Otherwise, I may take a look when I have some time to dedicate to Open Source.

**Comment by domihagen** [comment]
We have a similar/the same problem. We have created a scope function that takes care of sorting (`orderByPowerJoins`). This scope is used in several components that are included on a single page. Unfortunately the join is not inserted in all queries, which leads to an SQL error (Column not found).

**Comment by luisdalmolin** [comment]
Hey guys, I think this should be fixed by version 2.6.2: https://github.com/kirschbaum-development/eloquent-power-joins/releases/tag/2.6.2

If it doesn't, feel free to reopen the issue.

**Comment by jonneroelofs** [comment]
Thanks this works great! Was able to bin a bunch of workaround code I had to work around this problem.
